### PR TITLE
Support Guardian Quick crosswords

### PR DESCRIPTION
**Pull request by multimeric**
I didn't get time for the big rewrite I was hoping for, but I believe this works.

### CODE REVIEW

**Comment by stevenhao** [comment]
Surprisingly, using the copyright symbol in a string literal in source code is problematic (it was the last place I checked, in the process of debugging https://github.com/downforacross/downforacross.com/issues/90). This probably could've been avoided w/ a proper module bundler, so I'll try to get to setting that up for this project.

As part of the investigation / repro, I added `window.copyrightSymbol = "©";` to the source code, then inspected the popup -- it became `"Â©"`. I believe this is because the js source file is served with the incorrect mime type (or something along those lines -- unfortunately i'm not an expert in file encoding). I pushed a commit to fix this.

File: src/loaders/guardian.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        }
+
+        const title = doc.querySelector('h1').innerText;
+        const copyrightSymbol = String.fromCharCode(169); // cannot use inline © symbol in source code as it is served raw, and that messes up encoding
+        return {
+            "meta": {
+                "description": "",
+                "title": title,
+                "notes": "",
+                "author": "",
```
Comment (stevenhao) [inline src/loaders/guardian.js]: Is there any way to get the author? If not, perhaps just "The Guardian"?

File: src/loaders/guardian.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                }
+            }
+
+        }
+
+        const title = doc.querySelector('h1').innerText;
+        const copyrightSymbol = String.fromCharCode(169); // cannot use inline © symbol in source code as it is served raw, and that messes up encoding
+        return {
+            "meta": {
+                "description": "",
```
Comment (stevenhao) [inline src/loaders/guardian.js:59]: Any way to get the description?

**Review by stevenhao — CHANGES_REQUESTED** [review]
Ideally we can populate the description and author fields -- let me know if you can find a way to get these values!

File: src/loaders/guardian.js (hunk 1 above)
Comment (multimeric) [reply src/loaders/guardian.js]: It's quite strange, they actually don't provide it for [quick crosswords](https://www.theguardian.com/crosswords/quick/15587#4-down), but they do for [cryptics](https://www.theguardian.com/crosswords/cryptic/28118). So I'm not sure what to do about it. I'll put The Guardian as the author in the meantime.

File: src/loaders/guardian.js (hunk 2 above)
Comment (multimeric) [reply src/loaders/guardian.js:59]: Not really sure what to put there. If you have a look at one of their [quick crosswords](https://www.theguardian.com/crosswords/quick/15587#4-down), can you see anything that looks like a description? The title ("Quick crossword No 15,587") perhaps?

**Comment by multimeric** [comment]
You're right that the source of the problem was the perceived encoding of guardian.js, not anything else. But I solved that I think more neatly by just adding `<script src="loaders/guardian.js" charset="utf-8"></script>` to the HTML. When I view the extension source code in Chrome, it now no longer shows the wrong character.

File: src/loaders/guardian.js (hunk 2 above)
Comment (stevenhao) [reply src/loaders/guardian.js:59]: Ah, that's fine then. Description can be blank, not a big deal.

File: src/loaders/guardian.js (hunk 1 above)
Comment (stevenhao) [reply src/loaders/guardian.js]: > You're right that the source of the problem was the perceived encoding of guardian.js, not anything else. But I solved that I think more neatly by just adding `<script src="loaders/guardian.js" charset="utf-8"></script>` to the HTML. When I view the extension source code in Chrome, it now no longer shows the wrong character.

That's a much neater solution, thanks.

**Comment by stevenhao** [comment]
> You're right that the source of the problem was the perceived encoding of guardian.js, not anything else. But I solved that I think more neatly by just adding `<script src="loaders/guardian.js" charset="utf-8"></script>` to the HTML. When I view the extension source code in Chrome, it now no longer shows the wrong character.

That's a much neater solution, thanks.

**Comment by multimeric** [comment]
Thanks for the merge. Although I've found a small bug that's equally frustrating. The dashes in the clues are being converted to the paragraph symbol. For example: https://downforacross.com/beta/game/215914-smub. I'm really not sure where the error is occurring in this case.

**Comment by multimeric** [comment]
Can you publish a new version of the add-on with this change, please?

**Comment by stevenhao** [comment]
Yup, waiting for review!

**Comment by stevenhao** [comment]
@TMiguelT it's live!

**Comment by multimeric** [comment]
Cheers!

**Comment by stevenhao** [comment]
Closed #9

**Pull request merged**